README.md

```diff
@@ -1713,6 +1713,7 @@ end
 #### `length`
 
 Parameters with types that support `#length` method can be restricted to have a specific length with the `:length` option.
+In addition, if the received parameter value is `nil`, the length validation will not be triggered. If you do not allow `nil`, you can use the `allow_blank: false` option.
```

Comment:
Add "want to", so "If you do not want to allow..."

Reply:
um this will be problematic if you want to allow zero length parameters. For example 

```ruby
requires :some_list, type: [Integer], length: { min: 0, max: 10 }, allow_blank: false
requires :some_string, type: String, length: { min: 0 }, allow_blank: false
```

would both have validation errors if we provided the following parameters `{ some_list: [], some_string: '' }`

IMO this is incorrect behaviour.

Reply:
I think this is exactly the expected behavior. Why do you think this is incorrect @dhruvCW?

Reply:
I see like I mentioned combining the two can lead to unexpected results since allow_blank validates both nil as well as empty values. I personally then would appreciate this specific case to be documented so that users are aware about this potential pitfall.

Reply:
Yes, we should, @OuYangJinTing add specs for ^ and README pls?

Reply:
@dblock @dhruvCW Sorry, I'm a little busy with work these two days, and I didn't give timely feedback.
Based on this consideration, I think it should be better to add an allow_nil option to the `Grape::Validations::Validators::LengthValidator`. e.g:

```ruby
module Grape
  module Validations
    module Validators
      class LengthValidator < Base
        def initialize(attrs, options, required, scope, **opts)
          @min = options[:min]
          @max = options[:max]
          @allow_nil = options.key?(:allow_nil) ? options[:allow_nil] : true # allow `allow_nil` option, default to `true`

          super

          raise ArgumentError, 'min must be an integer greater than or equal to zero' if !@min.nil? && (!@min.is_a?(Integer) || @min.negative?)
          raise ArgumentError, 'max must be an integer greater than or equal to zero' if !@max.nil? && (!@max.is_a?(Integer) || @max.negative?)
          raise ArgumentError, "min #{@min} cannot be greater than max #{@max}" if !@min.nil? && !@max.nil? && @min > @max
        end

        def validate_param!(attr_name, params)
          param = params[attr_name]

          return if param.nil? && @allow_nil # when `allow_nil` is `true`, `nil` is allowed

          raise ArgumentError, "parameter #{param} does not support #length" unless param.respond_to?(:length)

          return unless (!@min.nil? && param.length < @min) || (!@max.nil? && param.length > @max)

          raise Grape::Exceptions::Validation.new(params: [@scope.full_name(attr_name)], message: build_message)
        end
        ...
end
```

If you agree, I will be happy to change it.

Reply:
I agree but I think the default should be `false`. maybe `ignore_nil` would be a better name ? since we are skipping the validation for `nil` values.

Reply:
@dhruvCW Thank you for your suggestions.
1. I agree that the default value be to false.
2. `ignore_nil` is a good name, but I use the name `allow_nil`, considering that there is already `allow_blank`, and it should be better to use a similar name.

Reply:
>ignore_nil is a good name, but I use the name allow_nil, considering that there is already allow_blank, and it should be better to use a similar name.

makes sense to keep the names consistent 😅 👍

Reply:
Yes, the other reason not to change these is to remain positive vs. negative.